x-pack/plugins/apm/public/components/routing/settings/index.tsx

```diff
@@ -141,6 +143,30 @@ export const settings = {
         element: <AgentKeys />,
         tab: 'agent-keys',
       }),
+      '/settings/agent-explorer': {
+        ...page({
+          title: i18n.translate(
+            'xpack.apm.views.settings.agentExplorer.title',
+            {
+              defaultMessage: 'Agent explorer',
+            }
+          ),
+          element: <AgentExplorer />,
+          tab: 'agent-explorer',
+        }),
+        params: t.type({
+          query: t.intersection([
+            environmentRt,
+            t.type({
+              kuery: t.string,
+            }),
+            t.partial({
+              agentLanguage: t.string,
```

Comment:
I would recommend making these required too. That means that we cannot forget about them.